### ISSUE TITLE
SEC-2865: Session URL rewriting being disabled should only strip JSESSIONID

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-security#3087]
[Rob Winch](https://jira.spring.io/secure/ViewProfile.jspa?name=rwinch) (Migrated from [SEC-2865](https://jira.spring.io/browse/SEC-2865?redirect=false)) said:

This is important so URL rewriting for something like Spring Session can take place

Things to consider:
- Users may have customized the COOKIE name (i.e. it may be a path variable other than JSESSIONID)
- Users may have path variables in their initial URL

**Comment by OrangeDog** [comment]
Just to clarify, it is known that by default `.sessionManagement()` disables ALL rewriting?
While technically correct, rather than "disallows HTTP sessions to be included in the URL", the documentation should be clear on this.

**Comment by jzheaux** [comment]
Good point, @OrangeDog. Would you like to submit a PR with some improved documentation around this?
